### PR TITLE
docs: Update top-level readme

### LINKED ISSUE #1362: Add top-level readme to monorepo
**Description**
Create a top-level monorepo readme that will shortly explain all projects of monorepo. 

**Acceptance Criteria**
- [ ] Readme document explaining all projects located in the monorepo repository. 

**Additional context**

### PR DESCRIPTION
**Pull request by solid-daniyilyevtyushkin**
Fixes #1362

## Checklist

- [x] Documented

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -1,104 +1,175 @@
...
+void main() {
+  group("TestClient", () {
+    final mockServer = MockServer();
+    TestClient client;
+
+    setUpAll(() async {
+      await mockServer.start();
+      client = TestClient(mockServer.url);
+    });
+
```
Comment (solid-vovabeloded) [inline readme.md:79]:
```suggestion
    tearDownAll(() async {
      await mockServer.close();
    });
```
Does it make sense?

File: readme.md [hunk 2]
```diff
@@ -1,104 +1,175 @@
...
+      },
+    );
+  });
+}
+```
+</details>
 
-Firebase Analytics is optional and may not be configured during deployment.
+## :shield: Guardian
+[Guardian](guardian) is a tool for detecting and publishing flaky tests by analyzing JUnit XML files. The tool accepts the unit test reports and compares the actual and previous test results stored in a database. If the test is considered flaky, Guardian notifies the team using Slack and creates an issue in Jira.
```
Comment (solid-vovabeloded) [inline readme.md:96]:
```suggestion
[Guardian](guardian) is a tool for detecting and notifying about flaky tests by analyzing JUnit XML files. This tool accepts the unit test reports and compares the actual and previous test results stored in a database. If the test is considered flaky, Guardian notifies the team using Slack and creates an issue in Jira.
```
What do you think? Also, would it be great to have a `Features` subsection in this section?

File: readme.md [hunk 3]
```diff
@@ -1,104 +1,175 @@
...
 
-### Safe
-
-All the data that the Metrics components orchestrate, belongs only to you: it is stored in the Cloud Firestore database as a part of your Firebase project, the logging is performed on your Sentry account, if enabled, analytics are reported to your Firebase Analytics account, and so on. Moreover, to make the components even safer, we provide a great set of [Firestore Security rules](https://firebase.google.com/docs/firestore/security/get-started) for Cloud Firestore to protect the stored projects' data. The [Firebase Authentication](https://firebase.google.com/docs/auth) stands on the guard of the deployed web application and the stored data from insufficient access.
-
-![Auth UI](docs/images/auth_ui.png)
-
-### Configurable
+## :world_map: YAML Map
+[YAML Map](yaml_map) is a wrapper around Dart's [`yaml`](https://pub.dev/packages/yaml) that simplifies working with YAML documents.
```
Comment (solid-vovabeloded) [inline readme.md:134]:
```suggestion
[YAML Map](yaml_map) is a wrapper around Dart's [`yaml`](https://pub.dev/packages/yaml) package that simplifies working with YAML documents.
```
Does it make sense?

File: readme.md [hunk 4]
```diff
@@ -1,104 +1,175 @@
-![Metrics Logo](docs/images/metrics_logo.png)
+# Getting Started
 
-Metrics is a set of software components to collect and review software project metrics like performance, build stability, and codebase quality.
+Consider this useful links that may help you to get started:
+1. [GitHub Agile process :chart_with_upwards_trend:](docs/02_process.md)
+2. [Dart code style :nail_care:](docs/09_dart_code_style.md)
+3. [Collaboration :raised_hands:](docs/10_collaboration.md)
```
Comment (solid-vovabeloded) [inline readme.md:6]: Would it be useful to add links to the `Effective Dart` and `process` document here?

File: readme.md [hunk 5]
```diff
@@ -1,104 +1,175 @@
-![Metrics Logo](docs/images/metrics_logo.png)
+# Getting Started
 
-Metrics is a set of software components to collect and review software project metrics like performance, build stability, and codebase quality.
+Consider this useful links that may help you to get started:
```
Comment (solid-vovabeloded) [inline readme.md:3]:
```suggestion
Consider these useful links that may help you to get started:
```
?

File: readme.md [hunk 6]
```diff
@@ -1,104 +1,175 @@
...
+```
 
-# Documentation :books:
+</details>
 
-You can find complete Metrics documentation using the following links:
-- [general documentation](docs/)
-- [CI integrations documentation](metrics/ci_integrations/docs/)
-- [Web documentation](metrics/web/docs/)
+# :scroll: License
```
Comment (solid-vovabeloded) [inline readme.md:174]: Would it be useful to add a link for the dependencies licenses document here? What do you think?

**Review by solid-vovabeloded — CHANGES_REQUESTED** [review]
Well done, @solid-daniyilyevtyushkin & @solid-yevheniiashchypeleva! I've added a couple of suggestions here - please take a look.

File: readme.md [hunk 7]
```diff
@@ -1,104 +1,187 @@
-![Metrics Logo](docs/images/metrics_logo.png)
+# Projects
+This repository holds the source code for the following projects:
```
Comment (solid-vovabeloded) [inline readme.md:2]:
```suggestion
This repository holds the source code of the following projects:
```
?

File: readme.md [hunk 8]
```diff
@@ -1,104 +1,187 @@
...
-
-### Fully Integrated
-
-The Metrics software components consist of a set of tools that provide a convenient way to deploy components, synchronize project data, configure logging, and collecting analytics. Moreover, the CI Integrations Tool can be integrated into the project's CI to automate builds synchronization and provide up-to-date data to the Web Metrics Application.
-
-![Concept map](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/platform-platform/monorepo/master/concept_map.puml)
+### Features
+- Authentication verification (by providing `AuthCredentials`);
+- Handling `GET`, `DELETE`, `POST`, `PUT` HTTP methods;
+- Flexible request URL matching (using `ExactPathMatcher`, `RegExpPathMatcher`).
```
Comment (solid-vovabeloded) [inline readme.md:17]:
```suggestion
- Flexible request URL matching (using `ExactPathMatcher` or `RegExpPathMatcher`).
```
Does it make sense?

File: readme.md [hunk 9]
```diff
@@ -1,104 +1,187 @@
...
+### Features:
+- Slack integration for notifications;
+- Parsing JUnit XML reports.
 
-#### Flutter Web
+## :bar_chart: Metrics
+[Metrics](metrics/readme.md) is a set of software components to collect and review software project metrics like performance, build stability, and codebase quality.
+The Metrics project includes the following components:
+- [Metrics Web](metrics/web) - a web application for the project metrics visualisation.
+- [CI Integrations](metrics/ci_integrations) - a CLI application that integrates with popular CI tools to collect software project metrics, such as Jenkins, GitHub Actions, and Buildkite.
```
Comment (solid-vovabeloded) [inline readme.md:105]:
```suggestion
- [CI Integrations](metrics/ci_integrations) - a CLI application that integrates with popular CI tools, such as Jenkins, GitHub Actions, and Buildkite, to collect software project metrics.
```
Would it be a bit more clear and readable?

File: readme.md [hunk 10]
```diff
@@ -1,104 +1,187 @@
...
+      },
+    );
+  });
+}
+```
+</details>
 
-A `Firebase` instance that provides the Firestore, Firebase Cloud Functions services and ability to deploy the application on Firebase Hosting. Also, provides an Analytics service used to gather and store the analytics data (this service is optional and may not be configured during deployment).
+## :shield: Guardian
+[Guardian](guardian) is a tool for detecting and notifying about flaky tests by analyzing JUnit XML files. This tool accepts the unit test reports and compares the actual and previous test results stored in a database. If the test is considered flaky, Guardian notifies the team using Slack and creates an issue in Jira.
```
Comment (solid-vovabeloded) [inline readme.md:95]:
```suggestion
[Guardian](guardian) is a tool designed for detecting and notifying about flaky tests by analyzing JUnit XML files. This tool accepts the unit test reports and compares the actual and previous test results stored in a database. If the test is considered flaky, Guardian notifies the team using Slack and creates an issue in Jira.
```
Does it make sense? What do you think?

File: readme.md [hunk 11]
```diff
@@ -1,104 +1,187 @@
...
+</details>
 
-Follow the next links to get acquainted with the Metrics project design: 
-- [design/](design/)
+# Getting Started
+Consider these useful links that may help you to get started:
+1. [GitHub Agile process :chart_with_upwards_trend:](docs/02_process.md)
+2. [Dart code style :nail_care:](docs/09_dart_code_style.md)
+3. [Collaboration :raised_hands:](docs/10_collaboration.md)
+4. [Effective Dart](https://dart.dev/guides/language/effective-dart)
```
Comment (solid-vovabeloded) [inline readme.md:182]:
```suggestion
4. [Effective Dart :dart:](https://dart.dev/guides/language/effective-dart)
```
What do you think?

File: readme.md [hunk 12]
```diff
@@ -1,104 +1,187 @@
...
 
-# Documentation :books:
+# :scroll: License
+Licensed under the terms of the Apache 2.0 License that can be found in the [LICENSE file](https://github.com/platform-platform/monorepo/blob/master/LICENSE).
 
-You can find complete Metrics documentation using the following links:
-- [general documentation](docs/)
-- [CI integrations documentation](metrics/ci_integrations/docs/)
-- [Web documentation](metrics/web/docs/)
+Consider the following [document](docs/15_dependencies_licenses.md) that describes the licenses for all 3-rd party libraries used in monorepo.
```
Comment (solid-vovabeloded) [inline readme.md:187]:
```suggestion
Consider the following [document](docs/15_dependencies_licenses.md) that describes the licenses for all 3-rd party libraries used in projects of this repository.
```
What do you think?

**Review by solid-vovabeloded — CHANGES_REQUESTED** [review]
Great job, @solid-daniyilyevtyushkin & @solid-yevheniiashchypeleva! I've added a few more suggestions here - please take a look.

File: readme.md [hunk 13]
```diff
@@ -1,104 +1,186 @@
...
-
-### Fully Integrated
-
-The Metrics software components consist of a set of tools that provide a convenient way to deploy components, synchronize project data, configure logging, and collecting analytics. Moreover, the CI Integrations Tool can be integrated into the project's CI to automate builds synchronization and provide up-to-date data to the Web Metrics Application.
-
-![Concept map](http://www.plantuml.com/plantuml/proxy?cache=no&fmt=svg&src=https://raw.githubusercontent.com/platform-platform/monorepo/master/concept_map.puml)
+### Features
+- Authentication verification (by providing `AuthCredentials`);
+- Handling `GET`, `DELETE`, `POST`, `PUT` HTTP methods;
+- Flexible request URL matching (using `ExactPathMatcher` or `RegExpPathMatcher`).
```
Comment (solid-vladlaba) [inline readme.md:17]: To make each item consistent with the pattern: 
```suggestion
The API Mock server provides the following functionality:

- Verifying requests' authentication (by providing `AuthCredentials`);
- Handling requests with `GET`, `DELETE`, `POST`, `PUT` HTTP methods;
- Matching requests' URL (using `ExactPathMatcher` or `RegExpPathMatcher`).
```
Or: 
```suggestion
The API Mock server allows mocking the following real-server functionality:

- Verify requests' authentication (by providing `AuthCredentials`);
- Handle [or Process (?)] requests with `GET`, `DELETE`, `POST`, `PUT` HTTP methods;
- Handle routing by matching requests' URL (using `ExactPathMatcher` or `RegExpPathMatcher`).
```

File: readme.md [hunk 14]
```diff
@@ -1,104 +1,186 @@
...
-#### Core
-
-A Dart package that provides a common classes to use within Metrics applications.
-
-#### Firebase
+  <summary>Usage example</summary>
+
+Consider this short example on how to use the API Mock Server.
+
+First, let's consider we have the following API client, and we need to test its `fetchBar` method:
```
Comment (solid-vladlaba) [inline readme.md:24]:
```suggestion
Consider this short example on how to use the API Mock Server.

Let's assume that we have the following API client with the `fetchBar` method we should cover with tests:
```

File: readme.md [hunk 15]
```diff
@@ -1,104 +1,186 @@
...
+      },
+    );
+  });
+}
+```
+</details>
 
-A `Firebase` instance that provides the Firestore, Firebase Cloud Functions services and ability to deploy the application on Firebase Hosting. Also, provides an Analytics service used to gather and store the analytics data (this service is optional and may not be configured during deployment).
+## :shield: Guardian
+[Guardian](guardian) is a tool designed for detecting and notifying about flaky tests by analyzing JUnit XML files. This tool accepts the unit test reports and compares the actual and previous test results stored in a database. If the test is considered flaky, Guardian notifies the team using Slack and creates an issue in Jira.
```
Comment (solid-vladlaba) [inline readme.md:95]:
```suggestion
[Guardian](guardian) is a tool designed for detecting flaky tests by analyzing JUnit XML reports and notifying the team about the results. This tool accepts the actual reports and compares them to the stored results in a database. If the test is considered flaky, Guardian notifies the team using Slack and/or Jira integrations.
```

File: readme.md [hunk 16]
```diff
@@ -1,104 +1,186 @@
...
-All the data that the Metrics components orchestrate, belongs only to you: it is stored in the Cloud Firestore database as a part of your Firebase project, the logging is performed on your Sentry account, if enabled, analytics are reported to your Firebase Analytics account, and so on. Moreover, to make the components even safer, we provide a great set of [Firestore Security rules](https://firebase.google.com/docs/firestore/security/get-started) for Cloud Firestore to protect the stored projects' data. The [Firebase Authentication](https://firebase.google.com/docs/auth) stands on the guard of the deployed web application and the stored data from insufficient access.
-
-![Auth UI](docs/images/auth_ui.png)
+## :world_map: YAML Map
+[YAML Map](yaml_map) is a wrapper around Dart's [`yaml`](https://pub.dev/packages/yaml) package that simplifies working with YAML documents.
 
-### Configurable
+### Features
+- Comparing to the [`yaml`](https://pub.dev/packages/yaml) package, parses the result to core Dart types.
+- Converting Dart Maps to YAML formatted strings.
```
Comment (solid-vladlaba) [inline readme.md:140]:
```suggestion
- Parsing the YAML documents to core Dart types.
- Converting Dart Maps to YAML formatted strings.
```

File: readme.md [hunk 17]
```diff
@@ -1,104 +1,186 @@
...
 
-# Documentation :books:
+# :scroll: License
+Licensed under the terms of the Apache 2.0 License that can be found in the [LICENSE file](https://github.com/platform-platform/monorepo/blob/master/LICENSE).
 
-You can find complete Metrics documentation using the following links:
-- [general documentation](docs/)
-- [CI integrations documentation](metrics/ci_integrations/docs/)
-- [Web documentation](metrics/web/docs/)
+Consider the following [document](docs/15_dependencies_licenses.md) that describes the licenses for all 3-rd party libraries used in projects of this repository.
```
Comment (solid-vladlaba) [inline readme.md:186]:
```suggestion
Consider the [Dependencies Licenses](docs/15_dependencies_licenses.md) document that describes the licenses for all 3-rd party libraries used in projects of this repository.
```
What do you think?

**Review by solid-vladlaba — CHANGES_REQUESTED** [review]
Well done, @solid-daniyilyevtyushkin @solid-yevheniiashchypeleva! I've added a couple of suggestions below - please let me know what you think!

**Review by solid-vladlaba — APPROVED** [review]
LGTM, thanks @solid-daniyilyevtyushkin @solid-yevheniiashchypeleva!:rocket:

**Review by solid-vovabeloded — APPROVED** [review]
LGTM, thanks @solid-daniyilyevtyushkin & @solid-yevheniiashchypeleva!

**Pull request merged**